intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/trace/VirtualFileBasedTrace.xtend

```diff
@@ -178,7 +184,8 @@ class VirtualFileBasedTrace extends AbstractTrace implements IIdeaTrace {
 	}
 
 	override IdeaModuleConfig getLocalProjectConfig() {
-		return new IdeaModuleConfig(getLocalProject());
+		val module = localProject
+		return workspaceConfigProvider.getWorkspaceConfig(module).findProjectByName(module.name)
```

Comment:
Using the first content root is bogus, because a file is not necessary belongs to it. We should use org.eclipse.xtext.idea.filesystem.IdeaWorkspaceConfig.findProjectContaining(URI) instead.